### PR TITLE
Refactor InsertFrameworks script to use MERGE

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story [AB#11203](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/11203)
Task [AB#12926](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12926)
Task [AB#12927](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12927)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql [hunk 1]
```diff
@@ -1,4 +1,21 @@
-﻿IF NOT EXISTS (SELECT * FROM dbo.Framework)
-    INSERT INTO dbo.Framework(Id, [Name], [Owner])
-    VALUES ('NHSDGP001', 'NHS Digital GP IT Futures Framework 1', 'NHS Digital');
+﻿DECLARE @frameworks AS TABLE
+(
+    [Id]          NVARCHAR (10)  NOT NULL PRIMARY KEY,
+    [Name]        NVARCHAR (100) NOT NULL,
+    [Owner]       NVARCHAR (100) NULL
+);
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql:6]: Can you update the formatting of the table variable to look like the below?

```sql
DECLARE @frameworks AS TABLE
(
    Id nvarchar(10) NOT NULL PRIMARY KEY,
    [Name] nvarchar(100) NOT NULL,
    [Owner] nvarchar(100) NULL
);
```

(We usually use lower case for data types and tend not to align the column names and data types. We only use square brackets where required too.)

File: src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+('DFOCVC001', 'Digital First Online Consultation and Video Consultation Framework 1', 'NHS England');
+
+MERGE INTO dbo.Framework AS TARGET
+    USING @frameworks AS SOURCE
+        ON TARGET.Id = SOURCE.Id
+    WHEN MATCHED THEN
+        UPDATE SET TARGET.[Name] = SOURCE.[Name]
+    WHEN NOT MATCHED BY TARGET THEN
+        INSERT (Id, [Name], [Owner])
+        VALUES (SOURCE.[Id], SOURCE.[Name], SOURCE.[Owner]);
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql:20]: Can you tweak the formatting here so it looks like this? (Just some tweaking to the spacing.)

```sql
MERGE INTO dbo.Framework AS TARGET
     USING @frameworks AS SOURCE ON TARGET.Id = SOURCE.Id
      WHEN MATCHED THEN
           UPDATE SET TARGET.[Name] = SOURCE.[Name]
      WHEN NOT MATCHED BY TARGET THEN
           INSERT (Id, [Name], [Owner])
           VALUES (SOURCE.[Id], SOURCE.[Name], SOURCE.[Owner]);
```

**Review by bjss-chris — CHANGES_REQUESTED** [review]
Looks good overall – just some small formatting changes requested.

File: src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql (hunk 1 above)
Comment (RahulJha-mastek) [reply src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql:6]: Formatted table query

File: src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql (hunk 2 above)
Comment (RahulJha-mastek) [reply src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertFrameworks.sql]: Formatted MERGE query

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**